### ISSUE TITLE
Move quest reward struct from quest struct to achievement struct

### ISSUE THREAD

**Issue by gabigarc03** [uchicago-cs/chiventure#1159]
What the title says. Have different quest endpoints give different rewards, allowing more flexibility in quest design. See the [current implementation wiki](https://github.com/uchicago-cs/chiventure/wiki/Quests-~-Current-Implementation) for more information.

**Comment by Castel-Jaloux** [comment]
Started making changes to the code to implement and reflect the new role of rewards in achievements as well as quests. Although the changes have been made, the tests still need to be updated to display accurate results.

**Comment by Castel-Jaloux** [comment]
Tests have been updated and run to ensure that no errors were spotted. After several iterations of editing, the errors were identified and addressed.

**Comment by Castel-Jaloux** [comment]
In this task, I changed the structure of the achievement struct to incorporate rewards within it, to allow for different paths on a quest's achievement tree to result in different rewards. After modifying the code as appropriate, I also edited the example files and the testing files to account for these changes, and ran the tests to ensure that there were no identifiable errors in the code. I then submitted a pull request, which is still under review. Further comments on this issue will be notated there.
